api/python/provisioner/__main__.py

```diff
@@ -114,18 +117,54 @@ def _set_logging(output_type, log_args=None):
             setattr(log_args, config.LOG_CONSOLE_HANDLER, False)
 
     # forcibly enable and configure rotation file logging
-    # for most important commands
-    if getattr(log_args, 'cmd', None) in config.LOG_FORCED_LOGFILE_CMDS:
+    #   - for most important commands
+    #   - for setup commands
+    # forcibly disable rsyslog
+    #   - for setup commands
+    cmd = getattr(log_args, 'cmd', None)
+    cmd_inst = commands.get(cmd)
+    if (
+        cmd in config.LOG_FORCED_LOGFILE_CMDS
+        or isinstance(cmd_inst, SetupCmdBase)
+    ):
+        if (
+            hasattr(log_args, config.LOG_RSYSLOG_HANDLER)
+            and isinstance(cmd_inst, SetupCmdBase)
+        ):
+            # disable rsyslog logging
+            setattr(log_args, config.LOG_RSYSLOG_HANDLER, False)
+
         if hasattr(log_args, config.LOG_FILE_HANDLER):
-            # enable
+            # enable file logging
             setattr(log_args, config.LOG_FILE_HANDLER, True)
             # set file to log if default value is set
             filename_attr = f'{config.LOG_FILE_HANDLER}_filename'
             if (
                 getattr(log_args, filename_attr) ==
                 attr.fields_dict(type(log_args))[filename_attr].default
             ):
-                filename = _generate_logfile_filename(log_args.cmd)
+                if isinstance(cmd_inst, SetupCmdBase):
```

Comment:
Separate logging fixes and design changes into different PRs.
Don't club the two as it becomes difficult to trace the changes.